input/elasticapm/internal/modeldecoder/v2/decoder.go

```diff
@@ -1549,6 +1543,10 @@ func mapToLogModel(from *log, event *modelpb.APMEvent) {
 	if len(from.Labels) > 0 {
 		modeldecoderutil.MergeLabels(from.Labels, event)
 	}
+	if event.Error == nil {
+		event.Event = populateNil(event.Event)
+		event.Event.Kind = "event"
+	}
```

Comment:
**question**: was this lost during the protobuf migration ? Is this something that we should backport to 8.9 ?

Reply:
No, I added this as a means to identify a document as being a log prior to its `data_stream.*` fields being set. No need to backport.